tests/application/cms/test_views.py

```diff
@@ -995,3 +949,139 @@ class TestChartBuilder(_TestVisualisationBuilder):
 
 class TestTableBuilder(_TestVisualisationBuilder):
     ROUTE_NAME = "cms.create_table"
+
+
+class TestRemoveDataSourceView:
+    def setup(self):
+        self.saved_config = {**current_app.config}
+
+    def teardown(self):
+        current_app.config = {**self.saved_config}
+
+    def test_login_required(self, test_app_client):
+        measure_version = MeasureVersionFactory()
+
+        res = test_app_client.post(
+            url_for(
```

Comment:
I decided against using the `url_for` helpers in my view tests – but I can see arguments for and against. Either way, _possibly_ it'd make the tests marginally more readable to refactor this to a shared helper method?

Reply:
I think we should be free to use `url_for` in tests. We could maybe have a single separate test that asserts `url_for` builds the exact URLs we expect. This then gives us assurance that we never break routing, and the flexibility to still use url_for so we aren't hardcoding URLs everywhere.

Reply:
But we definitely use `url_for` all over our tests at the moment, so I don't think we're at the point where we should suddenly stop using them without a plan in place to retire all the other uses, which I think is not worth doing.